spec/mixlib/install/product_spec.rb

```diff
@@ -0,0 +1,254 @@
+#
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+require "mixlib/install/product"
+
+context "Mixlib::Install::Product" do
+  context "for product_name when using strings" do
+    let(:product) {
+      Mixlib::Install::Product.new do
+        product_name "test-product"
+      end
+    }
+
+    it "accepts and returns the value correctly" do
+      expect(product.product_name).to eq("test-product")
+    end
+
+    it "returns nil for unset properties" do
+      expect(product.package_name).to eq(nil)
+    end
+
+    it "errors for unexisting properties" do
+      expect { product.address }.to raise_error(StandardError)
+    end
+  end
```

Comment:
Meh... Probably that would be a TODO that lives for a long time so I say let's not add it.  :smile: